java-security/src/main/java/com/sap/cloud/security/config/k8s/K8sEnvironment.java

```diff
@@ -14,76 +22,120 @@
 
 import static com.sap.cloud.security.config.Service.IAS;
 import static com.sap.cloud.security.config.Service.XSUAA;
+import static com.sap.cloud.security.config.k8s.K8sConstants.*;
 
 /**
  * Loads the OAuth configuration ({@link OAuth2ServiceConfiguration}) of a
  * supported identity {@link Service} in the Kubernetes Environment by
- * accessing defaults service secrets paths "/etc/secrets/sapcp/xsuaa" for Xsuaa service or "/etc/secrets/sapcp/ias" for .
- * IAS service
+ * accessing defaults service secrets paths "/etc/secrets/sapcp/xsuaa" for Xsuaa service
+ * or "/etc/secrets/sapcp/ias" for IAS service.
  */
 public class K8sEnvironment implements Environment {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(K8sEnvironment.class);
 
     private static K8sEnvironment instance;
-    private static final String DEFAULT_XSUAA_PATH = "/etc/secrets/sapcp/xsuaa";
-    private static final String DEFAULT_IAS_PATH = "/etc/secrets/sapcp/ias";
-    private static String customXsuaaPath;
-    private static String customIasPath;
+    private String xsuaaPath = DEFAULT_XSUAA_PATH;
+    private String iasPath = DEFAULT_IAS_PATH;
+    private String serviceManagerPath = DEFAULT_SERVICE_MANAGER_PATH;
+    private static CloseableHttpClient httpClient;
 
-    private Map<Service, List<OAuth2ServiceConfiguration>> serviceConfigurations;
-    private UnaryOperator<String> systemEnvironmentProvider;
-    private UnaryOperator<String> systemPropertiesProvider;
+    private final Map<Service, Map<String, OAuth2ServiceConfiguration>> serviceConfigurations = new HashMap<>(); // use getServiceConfigurations() instead
+    private OAuth2ServiceConfiguration serviceManagerConfigurations;
+    private static OAuth2ServiceManagerService serviceManagerService;
 
-    public static K8sEnvironment getInstance() {
-        return getInstance(System::getenv, System::getProperty);
-    }
+    private K8sEnvironment() {}
 
-    public static K8sEnvironment getInstance(UnaryOperator<String> systemEnvironmentProvider,
-                                            UnaryOperator<String> systemPropertiesProvider) {
+    public static K8sEnvironment getInstance() {
         if (instance == null) {
             instance = new K8sEnvironment();
+            instance.serviceManagerConfigurations = loadServiceManagerConfig(instance.serviceManagerPath);
         }
-        instance.systemEnvironmentProvider = systemEnvironmentProvider;
-        instance.systemPropertiesProvider = systemPropertiesProvider;
-        instance.serviceConfigurations = loadAll();
         return instance;
     }
 
-    public static K8sEnvironment getInstance(@Nullable String customXsuaaPath, @Nullable String customIasPath) {
-        K8sEnvironment.customXsuaaPath = customXsuaaPath;
-        K8sEnvironment.customIasPath = customIasPath;
-        return getInstance(System::getenv, System::getProperty);
+    public K8sEnvironment withXsuaaPath(String customXsuaaPath) {
```

Comment:
I've removed the configuration of pathes and client from constructor arguments into setters, but still dislike it.
See i want to call Environment.getCurrent() all the time and do not like to redefine pathes and httpclient again and again. So, I guess we need a different approach (for pathes we can use SystemEnv / using again systemEnvironmentProvider and systemPropertiesProvider)....